### PR TITLE
[CONTRIB] Updating Special Characters Expectation

### PR DESCRIPTION
**Pull request by calvingdu**
Made a change to **expect_column_values_to_not_contain_special_characters.py** to accept a variable **allowed_characters** that acts as a list of characters that will not be validated against in this expectation

Also added spark implementation 

- [] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [X] Appropriate tests and docs have been updated

### CODE REVIEW

**Comment by calvingdu** [comment]
@cla-bot check

**Comment by TrangPham** [comment]
Thanks for your contributions :) I have taken a look at the code and kicked off the CI pipeline for you. The code looks good to me. @Kilo59 I wanted to confirm that the example test runner would automatically start running the test cases for spark now that the spark implemented is added - is this true?

File: contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_not_contain_special_characters.py [hunk 1]
```diff
@@ -31,12 +36,16 @@ class ColumnValuesToNotContainSpecialCharacters(ColumnMapMetricProvider):
...
+    condition_value_keys = ("allowed_characters",)
 
     # This method defines the business logic for evaluating the metric when using a PandasExecutionEngine
     @column_condition_partial(engine=PandasExecutionEngine)
-    def _pandas(cls, column, **kwargs):
+    def _pandas(cls, column, allowed_characters: list or set, **kwargs):
         def not_contain_special_character(val, *special_characters):
+            special_characters = [
+                char for char in special_characters if char not in allowed_characters
+            ]
```
Comment (Shinnnyshinshin) [inline contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_not_contain_special_characters.py:47]: 👍🏼

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you very much @calvingdu. This is a great change 🙇🏼

**Comment by Shinnnyshinshin** [comment]
> Thanks for your contributions :) I have taken a look at the code and kicked off the CI pipeline for you. The code looks good to me. @Kilo59 I wanted to confirm that the example test runner would automatically start running the test cases for spark now that the spark implemented is added - is this true?

Hi @TrangPham I saw your message from before and wanted to confirm that the Spark tests are running as part of CI/CD now. I'm on point for Link this week, so I'll make sure this gets in.

**Pull request merged**